### PR TITLE
CombSort.java

### PR DESCRIPTION
**Pull request by sandeeproy99**
Comb Sort is mainly an improvement over Bubble Sort. Bubble sort always compares adjacent values. So all inversions are removed one by one. Comb Sort improves on Bubble Sort by using gap of size more than 1. The gap starts with a large value and shrinks by a factor of 1.3 in every iteration until it reaches the value 1. Thus Comb Sort removes more than one inversion counts with one swap and performs better than Bubble Sort.

The shrink factor has been empirically found to be 1.3 (by testing CombSort on over 200,000 random lists) [Source: Wiki]

Although, it works better than Bubble Sort on average, worst case remains O(n2).

### CODE REVIEW

File: Sorts/CombSort.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
+// Java program for implementation of Comb Sort
+
+class CombSort
+{
+    // To find gap between elements
+    int getNextGap(int gap)
+    {
+        // Shrink gap by Shrink factor
+        gap = (gap*10)/13;
+        if (gap < 1)
```
Comment (christianbender) [inline Sorts/CombSort.java]: Use curly braces. You can remove the ```if``` complete and writing instead ```gap = (gap < 1 ) ? 1 : gap;```

File: Sorts/CombSort.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ 
+    // Driver method
+    public static void main(String args[])
+    {
+        CombSort ob = new CombSort();
+        int arr[] = {8, 4, 1, 56, 3, -44, 23, -6, 28, 0};
+        ob.sort(arr);
+ 
+        System.out.println("sorted array");
+        for (int i=0; i<arr.length; ++i)
```
Comment (christianbender) [inline Sorts/CombSort.java]: Use curly braces

File: Sorts/CombSort.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        int n = arr.length;
+ 
+        // initialize gap
+        int gap = n;
+ 
+        // Initialize swapped as true to make sure that loop runs
+        boolean swapped = true;
+ 
+        // Keep running while gap is more than 1 and last iteration caused a swap
+        while (gap != 1 || swapped == true)
```
Comment (christianbender) [inline Sorts/CombSort.java]: You can write ```swapped``` instead of ```swapped == true```

File: Sorts/CombSort.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
+// Java program for implementation of Comb Sort
+
+class CombSort
+{
+    // To find gap between elements
+    int getNextGap(int gap)
```
Comment (christianbender) [inline Sorts/CombSort.java]: You can this methods declare as ```static```. If you do this you can the methods right away use in the main-method.

File: Sorts/CombSort.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    {
+        // Shrink gap by Shrink factor
+        gap = (gap*10)/13;
+        if (gap < 1)
+            return 1;
+        return gap;
+    }
+ 
+    // Function to sort arr[] using Comb Sort
+    void sort(int arr[])
```
Comment (christianbender) [inline Sorts/CombSort.java]: You can this methods declare as ```static```. If you do this you can the methods right away use in the main-method.

File: Sorts/CombSort.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+                    swapped = true;
+                }
+            }
+        }
+    }
+ 
+    // Driver method
+    public static void main(String args[])
+    {
+        CombSort ob = new CombSort();
```
Comment (christianbender) [inline Sorts/CombSort.java]: Redundant: see my comments above.

**Comment by sandeeproy99** [comment]
Thank You. Done all the changes.

**Pull request merged**